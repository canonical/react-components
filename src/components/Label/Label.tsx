import classNames from "classnames";
import React from "react";
import type { LabelHTMLAttributes, ReactNode } from "react";

import type { ClassName, PropsWithSpread } from "types";

/**
 * The props for the Label component.
 */
export type Props = PropsWithSpread<
  {
    /**
     * The label content.
     */
    children?: ReactNode;
    /**
     * Optional class(es) to give to the label element.
     */
    className?: ClassName;
    /**
     * The id of the input this label is connected to.
     */
    forId?: string;
    /**
     * Whether to apply required styling to the label.
     */
    required?: boolean;
  },
  LabelHTMLAttributes<HTMLLabelElement>
>;

const Label = ({
  children,
  className,
  forId,
  required,
  ...props
}: Props): JSX.Element => (
  <label
    className={classNames(className, "p-form__label", {
      "is-required": required,
    })}
    htmlFor={forId}
    {...props}
  >
    {children}
  </label>
);

export default Label;
