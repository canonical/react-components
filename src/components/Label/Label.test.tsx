import { render, screen } from "@testing-library/react";
import React from "react";

import Label from "./Label";

describe("Label ", () => {
  it("renders", () => {
    render(
      <Label forId="test-id" data-testid="label">
        Test content
      </Label>
    );
    expect(screen.getByTestId("label")).toMatchSnapshot();
  });

  it("can be required", () => {
    render(<Label required={true} data-testid="label" />);
    expect(screen.getByTestId("label")).toHaveClass("is-required");
  });

  it("can add additional classes", () => {
    render(<Label className="extra-class" data-testid="label" />);
    const label = screen.getByTestId("label");
    expect(label).toHaveClass("p-form__label");
    expect(label).toHaveClass("extra-class");
  });
});
