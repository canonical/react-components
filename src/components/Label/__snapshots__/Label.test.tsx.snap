// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label  renders 1`] = `
<label
  className="p-form__label"
  htmlFor="test-id"
>
  Test content
</label>
`;
